src/python/pants/pantsd/process_manager.py

```diff
@@ -499,3 +499,57 @@ def post_fork_child(self):
 
   def post_fork_parent(self):
     """Post-fork parent callback for subclasses."""
+
+
+class FingerprintedProcessManager(ProcessManager):
+  """A `ProcessManager` subclass that provides a general strategy for process fingerprinting."""
+
+  FINGERPRINT_KEY = 'fingerprint'
+  FINGERPRINT_CMD_KEY = None
+  FINGERPRINT_CMD_SEP = '='
+
+  @property
+  def fingerprint(self):
+    """The fingerprint of the current process.
+
+    :returns: The fingerprint of the running process as read from the process table, ProcessManager
```

Comment:
It's not clear why this is an OR... a bit more information about why we would check X then Y would be good.

Reply:
added a comment in the docstring.